mlflow/store/artifact/runs_artifact_repo.py

```diff
@@ -128,7 +131,27 @@ def download_artifacts(self, artifact_path, dst_path=None):
         Returns:
             Absolute path of the local filesystem location containing the desired artifacts.
         """
-        return self.repo.download_artifacts(artifact_path, dst_path)
+        try:
+            return self.repo.download_artifacts(artifact_path, dst_path)
+        except Exception:
+            _logger.debug(
+                "Failed to download artifacts. "
+                "Attempting to download from the model associated with the run."
+            )
+            run_id = RunsArtifactRepository.parse_runs_uri(self.artifact_uri)[0]
+            client = mlflow.tracking.MlflowClient()
+            run = client.get_run(run_id)
+            # Search for models associated with the run
+            models = client.search_logged_models(
+                experiment_ids=[run.info.experiment_id],
+                filter_string=f"run_id = {run_id}",
+            )
+            if len(models) != 1:  # no model or multiple models found
+                raise
```

Comment:
There could totally be multiple logged models with the same source run ID. We shouldn't raise.

Instead, we should filter by name - the name should match the artifact path. If there are multiple models with the same name and source run ID, we should print a warning and pick the first one.

Reply:
Sounds good!